### DISCUSSION TITLE
Plugin for the future image formats, ex. AVIF

### DISCUSSION THREAD

**Discussion by o-alexandrov** [ionic-team/capacitor#4026] [Q&A]
Hello, I'm new to Capacitor and it would be great to know your experience.
If I needed to add support for future image formats, like AVIF that's not widely supported in browsers, how would I do so with Capacitor?

As I understand, it's possible by creating a plugin.
I searched for such, seemingly very useful for the community plugin, but had no results, maybe you know one that exists but has difficult keywords that make it hard to find?

**Comment by jcesarmobile** [comment]
what do you want to do it with those images?

If you want to show them in an `img` tag or canvas, then that support should be added by the WebView owners (Google for Android, Apple for iOS).
If you want to show them in native views, then you can probably do it with a plugin.

**Reply by o-alexandrov** [reply]
Third option you didn't mention is to convert the newer formats to browser supported ones and then present them in the web UI.
It should perform quick for a few images at a time, but might become slow for infinite list of images.

I am looking for this third option to convert images in the native layer, before returning them to the web layer.

**Reply by o-alexandrov** [reply]
It would be great if there was a plugin that would expose API of:
- [libvips](https://github.com/libvips/libvips)
- or if there is a procedure of adding node based plugins in Capacitor.js, it would be great to expose [sharp's](https://github.com/lovell/sharp) API, which in turn calls libvips

**Comment by gelinger777** [comment]
you can use for image backend awesome  tool named imgproxy . It has a cool feature named  IMGPROXY_ENABLE_WEBP_DETECTION . What it does is it detects via the header if browser(webview) supports webp/avif and serves them, if no it falls back to jpeg/png. So when time comes and webview support them, then you will be onboard automatically
